docs/usage/security-and-permissions.md

```diff
@@ -65,6 +65,101 @@ For example, if you have an `npm` package and do not configure a private registr
 You could avoid this by configuring private registries but such registries need to query public registries anyway.
 We don't know of any public registries which reverse lookup IP addresses to associate companies with packages.
 
+#### Security advisory for self-hosted Renovate instances
+
+##### Introduction
+
+Before you start self-hosting Renovate you must understand the security implications associated with monitoring and updating repositories.
+The process that Renovate uses to update dependencies runs under the same user context as the Renovate process itself.
+This means it (what does it refer to here???) has the same level of access to information and resources as the user context!
+
+##### Trusting Repository Developers
+
+All self-hosted Renovate instances must operate under a trust relationship with the developers of the monitored repositories.
+This has the following implications:
+
+- Access to information
+- Execution of code
+
+Keep reading to learn more.
+
+###### Access to information
+
+Since the update process runs with the _same_ user privileges as the Renovate process, it inherently has access to the same information and resources.
+This includes sensitive data that may be stored within the environment where Renovate is hosted.
+
+###### Execution of code
+
+In certain scenarios, code from the monitored repository is executed as part of the update process.
+This is particularly true during `postUpgradeTasks`, where scripts specified by the repository are run.
```

Comment:
I think we should add, that this is not restricted to the postUpgradeTasks, for instance when a wrapper within the repo is called like `gradlew`

Reply:
```suggestion
This is particularly true during, for example:

- `postUpgradeTasks`, where scripts specified by the repository are run
- when a wrapper within the repository is called, like `gradlew`

```